casting/converter.go

```diff
@@ -0,0 +1,110 @@
+package casting
+
+import (
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func CastSingleElement(inputTypeMap map[string]string, inputName string, inputData interface{}) interface{} {
```

Comment:
I was thinking that this function and the one below might be a struct/object, with the config `inputTypeMap map[string]string` as state, what do you think?
Then calling the conversion will require less parameters.

Reply:
Yes, I agree, still need to apply this

Reply:
Done

Reply:
Not sure that is done, I cannot see any struct here. However I can open a next issue for this minimal refactoring later.